### DISCUSSION TITLE
Own database update

### DISCUSSION THREAD

**Discussion by vhmoura** [osm-search/Nominatim#2108] [General]
Hi,

I have my own nominatim/postgres build machines and I would like to understand the best way to update the data in the database. In my case, I am interested in updating the roadspeed tag of an osm_id. What is the easiest way to accomplish that?

Many thanks.
Vini

**Comment by lonvia** [comment]
The best way of updating data is of course to participate in OpenStreetMap and improve the data there. You can then [apply updates](https://nominatim.org/release-docs/latest/admin/Update/) to import the changes into your Nominatim database.

If that is not an option for some reason and you want to add private data, then information like roadspeed can simply be added to the `extratags` column in the main `placex` table. This kind of extra information is not used by Nominatim in any way except to return it to the user. The only exceptions are (at the time of writing): linked_place, wikipedia and wikidata. So you can add information as you like. Just one catch: you have to disable triggers on the table before doing updates and reenable them afterwards. Also, don't forget that the Nominatim database is under an ODbL license, so the usual restrictions for added private data apply.

**Reply by vhmoura** [reply]
Thanks for your feedback Lonvia. You are right, it is much easier to just contribute which I am super happy to do so. This will avoid the hassle of updating tables, dealing with triggers when in fact the end result would be the same. 
For and OBdL point of view - if I understood it correctly - does the license text needs to be visible somewhere (external portal-internal portal)? What I am planning is to have is my own internal nomatim version (linux or docker) and primarily do reverse geocoding to gather street name, town etc and dump in another DB. Let me know how to deal with that. 
Cheers.

**Reply by lonvia** [reply]
I've just mentioned the license here, because ODbL is share-alike. If you would mix in your own private data with a Nominatim database, your data becomes subject to the ODbL. It doesn't matter much when you use the database only internally. But is something you have to be aware of, when you publish or share the database in any form. But as you are happy to contribute to OSM directly, there is no issue at all.

**Reply by vhmoura** [reply]
Hi Lonvia,
Just another question regarding updating the maps.
In my case the idea is to use either a GPS device or an mobile SDK (still to be be decided) to collect latitude/longitude and speed so we can determine what the road speed would be. With that data in hands then use the osmid from that location to update street maps roads that have zero road speed. 
My company is more than happy to use this data and contribute, that is not an issue at all. My concern is if I can use the data from a small fleet of vehicle/devices or if I need to be "sitting and driving a vehicle myself" to be able to use the data.
Any advice would be extremely helpful.
